src/cmd_kcl.rs

```diff
@@ -826,6 +828,94 @@ impl crate::cmd::Command for CmdKclSurfaceArea {
     }
 }
 
+/// Lint a KCL file for style issues.
+///
+///     # check a file for issues
+///     $ zoo kcl lint my-file.kcl
+///
+///     # pass a file from stdin
+///     $ cat my-file.kcl | zoo kcl lint -
+#[derive(Parser, Debug, Clone)]
+#[clap(verbatim_doc_comment)]
+pub struct CmdKclLint {
+    /// The path to the input file.
+    /// If you pass `-` as the path, the file will be read from stdin.
+    #[clap(name = "input", required = true)]
+    pub input: std::path::PathBuf,
+
+    /// Print a long-form description of what the issue is, and the rational
+    /// behind why.
+    #[clap(long, default_value = "false")]
+    pub descriptions: bool,
+
+    /// Show where the offending KCL source code is.
+    #[clap(long, short, default_value = "false")]
+    pub show_code: bool,
+}
+
+#[async_trait::async_trait(?Send)]
+impl crate::cmd::Command for CmdKclLint {
+    async fn run(&self, ctx: &mut crate::context::Context) -> Result<()> {
+        let path = self.input.to_str().unwrap_or("");
+        let input = ctx.read_file(path)?;
+        let input = std::str::from_utf8(&input)?;
+
+        // Parse the file.
+        let tokens = kcl_lib::token::lexer(input)?;
+        let parser = kcl_lib::parser::Parser::new(tokens);
+        let program = parser.ast()?;
+
+        for discovered_finding in program.lint(kcl_lib::lint::checks::lint_variables)? {
```

Comment:
I made a lint_all because I added another rule for a backwards incompatible future change 😇